plugins/org.python.pydev/tests_analysis/com/python/pydev/analysis/OccurrencesAnalyzerPy310Test.java

```diff
@@ -81,4 +81,330 @@ public void testMatchAndCaseAsSoftKeyword() {
+
+    public void testMatchStmt4() {
+        doc = new Document(""
+                + "class Point:\n"
+                + "    def __init__(self, x, y):\n"
+                + "        self.x = x\n"
+                + "        self.y = y\n"
+                + "def foo(points):\n"
+                + "    x = 0\n"
+                + "    y = 0\n"
+                + "    y1 = 1\n"
+                + "    y2 = 2\n"
+                + "    match points:\n"
+                + "        case []:\n"
+                + "            print(\"No points\")\n"
+                + "        case [Point(0, 0)]:\n"
+                + "            print(\"The origin\")\n"
+                + "        case [Point(x, y)]:\n"
+                + "            print(f\"Single point {x}, {y}\")\n"
+                + "        case [Point(0, y1), Point(0, y2)]:\n"
+                + "            print(f\"Two on the Y axis at {y1}, {y2}\")\n"
+                + "        case _:\n"
+                + "            print(\"Something else\")");
+        checkNoError();
+    }
+
+    public void testMatchStmt5() {
+        doc = new Document(""
+                + "class Point:\n"
+                + "    def __init__(self, x, y):\n"
+                + "        self.x = x\n"
+                + "        self.y = y\n"
+                + "def foo(point):\n"
+                + "    x = 0\n"
+                + "    y = 1\n"
+                + "    match point:\n"
+                + "        case Point(x, y) if x == y:\n"
+                + "            print(f\"Y=X at {x}\")\n"
+                + "        case Point(x, y):\n"
+                + "            print(f\"Not on the diagonal\")");
+        checkNoError();
+    }
+
+    public void testMatchStmt6() {
+        doc = new Document(""
+                + "class Color:\n"
+                + "    RED = 0\n"
+                + "    GREEN = 1\n"
+                + "    BLUE = 2\n"
+                + "def foo(color):\n"
+                + "    match color:\n"
+                + "        case Color.RED:\n"
+                + "            print(\"I see red!\")\n"
+                + "        case Color.GREEN:\n"
+                + "            print(\"Grass is green\")\n"
+                + "        case Color.BLUE:\n"
+                + "            print(\"I'm feeling the blues :(\")");
+        checkNoError();
+    }
+
+    public void testWildcardPattern() {
+        doc = new Document(""
+                + "match \"foo bar\".split()\n"
+                + "    case (\"doo\", \"lee\"):\n"
+                + "        pass\n"
+                + "    case _:\n"
+                + "        pass");
+        checkNoError();
+    }
+
+    public void testNonPatternWildcard() {
+        doc = new Document(""
+                + "x = 10\n"
+                + "match \"foo bar\".split()\n"
+                + "    case (\"doo\", \"lee\"):\n"
+                + "        pass\n"
+                + "x = _");
+        checkError("Undefined variable: _");
```

Comment:
The idea here is that there was a `case _:` and the `x = _` was made inside of that case...